apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs

```diff
@@ -268,5 +246,42 @@ internal void WriteDiagnostics(TextWriter writer)
             writer.WriteLine(Invariant($"{DateTime.UtcNow:yyyy-MM-dd'T'HH:mm:ss} - GoogleLogger will write logs to: {GetGcpConsoleLogsUrl()}"));
             writer.Flush();
         }
+
+        /// <summary>
+        /// Obtains the current ambient scope, which is not set by user code
+        /// but instead it is calculated based on the Logger configuration.
+        /// The ambient scope will be applied to all log entries.
+        /// It will be applied before the user specified scopes so that the
+        /// user code is able to override ambient scope values on a per
+        /// log entry basis.
+        /// </summary>
+        internal class AmbientScopeManager
+        {
+            private readonly GoogleLoggerScope _permanentParent;
+            private readonly IServiceProvider _serviceProvider;
+
+            internal AmbientScopeManager(LoggerOptions options, IServiceProvider serviceProvider)
+            {
+                _permanentParent = options?.Labels is null ? null : GoogleLoggerScope.CreateScope(new LabellingScopeState(options.Labels), null);
+                _serviceProvider = serviceProvider;
+            }
+
+            public GoogleLoggerScope GetCurrentScope()
```

Comment:
Presumably most of the time this will return null really quickly? Otherwise I can see the inefficiencies could be a little alarming, given that it's invoked for every log entry.

Reply:
Changed to use a single dictionary and labelling scope.